### PR TITLE
16 restaurant page UI

### LINKED ISSUE #16: Restaurant Page
Allow users to select any restaurant, and then browse all the menu items.

This solves the use case of going to a specific restaurant and trying to make an informed decision on what to order.

### PR DESCRIPTION
**Pull request by abraham-rostampoor**
* Created RestaurantsPage.js UI

### CODE REVIEW

**Review by mehradadimi — CHANGES_REQUESTED** [review]
Looks great @abraham-rostampoor , thank you!
A few changes come to my mind to be made:

1) The width of the search bar exceeds the width of the content below, maybe we can reduce that to be exactly the same as the width of the content below it?
Screenshot 2024-03-22 at 1 44 40 PM

2) I notices that the box shadow when hovered on the box does not show up in dark mode, its good to have that rolling.

3) The `width` of the search bar changes as we search for restaurants, which MAY come from the search bar and boxes below it being in the same container??

4) When the search does not find a restaurant, it leaves the page empty. Ideally, in that case, we want the page to show a message like "Sorry, we do not have that restaurant yet". or sth like that.

5) I think we need a little bit more distance from the hamburger icon and the boxes.

Screenshot 2024-03-22 at 1 54 06 PM


6) Please fix the Navbar regression. Not sure where it comes from, maybe this PR or others. I didn't see it on Johns PR.
Screenshot 2024-03-22 at 1 55 58 PM

**Comment by johnmck95** [comment]
How's this coming along @abraham-rostampoor, do you need any help?  Wondering if you've set it up to render the list of menu items once you click on a specific restaurant?

**Comment by abraham-rostampoor** [comment]
## Overview
This pull request introduces a series of updates that significantly improve the restaurant search functionality and overall user experience on the platform. It notifiably enhances the UI consistency and interactivity, refines the navigation flow, and addresses feedback regarding the search feature's visual and functional aspects.

## Changes Made

### RestaurantsPage.js Updates
- Implemented conditional rendering logic to ensure that the "No meals found" message is only displayed when appropriate, specifically when a restaurant is selected but returns no meals.
- Optimized the search bar to match the width of the content below, ensuring a cohesive and aesthetically pleasing layout.
- Fixed an issue in dark mode where the box shadow on hover was not visible, enhancing UI feedback and interactivity.
- Ensured the width of the search bar remained consistent during the search process to maintain a clean and stable page layout.
- Introduced a friendly message to inform users when a search yields no results, improving the user experience during restaurant discovery.
- Adjusted the spacing between the navigation elements and the content to enhance readability and visual appeal.

### New Navigation Flow
- Developed `MealsPage.js` to display restaurant-specific meal options on a separate page, improving the site's navigational structure and user engagement.
- Integrated a "Back to Restaurants" button with consistent styling to the "Show Meals" button, including a colour change on hover for better visual cues.
- Standardized the size of the displayed food items, mirroring the layout found on the macro page and adjusting dynamically to screen size for responsive design compatibility.

### UI Enhancements
- Updated hover effects and interactive elements across the platform to be more consistent with user expectations and modern design standards.
- Improved the visual integration of restaurant logos and ensured better error handling for missing images, creating a smoother visual experience.
- Refined spacing and alignment across components for a polished, professional look and feel.

File: nutridine/src/hooks/useNutrientPreferences.js [hunk 1]
```diff
@@ -21,7 +21,9 @@ const useNutrientPreferences = (userUid) => {
           nutrientPreferencesRef,
           where("userUid", "==", userUid)
         );
+        console.log(queryByUser);
```
Comment (johnmck95) [inline nutridine/src/hooks/useNutrientPreferences.js]: can we remove these 2 logs please?

**Review by johnmck95 — CHANGES_REQUESTED** [review]
Looks great! The only issues I came cross is the search bar on mobile spans the entire width, and should probably have some margin on all sides, and the error message seen here doesn't fit on the screen

image

**Review by mehradadimi — APPROVED** [review]

**Comment by johnmck95** [comment]
@abraham-rostampoor the macro and restaurant pages aren't protected routes anymore. Can you please fix that then feel free to merge

**Review by johnmck95 — APPROVED** [review]

**Pull request merged**